Custom status feature

chetanyakan opened the pull request:
#### Summary
- Create custom status components.
    - Custom status option in status modal
    - Custom status modal
- Actions to set/unset status.

#### Related Pull Requests
- https://github.com/brightscout-alpha/mattermost-redux/pull/3

#### Screenshots

chetanyakan committed:
Webapp config setting

jatinkksharma committed:
added custom status modal component and action to set custom status

jatinkksharma committed:
udpate the styles for custom status modal

jatinkksharma committed:
MI-1109 Update the logic to render emoji

jatinkksharma committed:
show prefilled custom status in modal

jatinkksharma committed:
update the drowdown component

jatinkksharma committed:
add logic to unset custom status

chetanyakan committed:
CSS Fixes

chetanyakan committed:
fix lint

jatinkksharma committed:
add logic to clear custom status

jatinkksharma committed:
Add emoji in legacy sidebar

jatinkksharma committed:
add a component for custom status suggestion

jatinkksharma committed:
update styles for legacy siderbar

jatinkksharma committed:
update the ui for recent custom status suggestions

jatinkksharma committed:
add logic to remove recent custom status

chetanyakan committed:
Remove default title in custom status emoji

chetanyakan committed:
Merge pull request #2 from brightscout-alpha/MI-1006

Add webapp config setting

jatinkksharma committed:
add custom status type in store

jatinkksharma committed:
remove aria-hidden attribute

chetanyakan committed:
Merge pull request #4 from brightscout-alpha/MI-1109_1

MI-1109 Add logic to show recent custom statuses

chetanyakan committed:
Merge branch 'MI-1109' of github.com:brightscout-alpha/mattermost-webapp into MI-1109_2

* 'MI-1109' of github.com:brightscout-alpha/mattermost-webapp:
  remove aria-hidden attribute
  add custom status type in store

chetanyakan committed:
ReFactor to custom component

jatinkksharma committed:
misc review fixes

jatinkksharma committed:
fix object immutable issue

jatinkksharma committed:
remove user reducer

chetanyakan committed:
Merge pull request #3 from brightscout-alpha/MI-1109

MI-1109 Added modal component and actions to set custom status

chetanyakan committed:
Merge branch 'custom-status' of github.com:brightscout-alpha/mattermost-webapp into MI-1109_2

* 'custom-status' of github.com:brightscout-alpha/mattermost-webapp:
  remove user reducer
  fix object immutable issue
  misc review fixes
  Webapp config setting

jatinkksharma committed:
add clear button in status dropdown

chetanyakan committed:
Use custom status emoji component

chetanyakan committed:
Code refactoring

chetanyakan committed:
Merge pull request #6 from brightscout-alpha/MI-1109_2

Custom status emoji component

jatinkksharma committed:
Merge branch 'custom-status' of github.com:brightscout-alpha/mattermost-webapp into MI-1135

jatinkksharma committed:
revert package-lock changes

jatinkksharma committed:
Merge pull request #5 from brightscout-alpha/MI-1135

Added button to clear custom status in status dropdown

jatinkksharma committed:
add custom status in user profile popover

chetanyakan committed:
Emoji icon CSS Fixes

jatinkksharma committed:
Merge branch 'icon' of github.com:brightscout-alpha/mattermost-webapp into MI-1136

jatinkksharma committed:
update styles for custom emoji

jatinkksharma committed:
fix css for channel header

chetanyakan committed:
ADD : Custom status in multiple places

chetanyakan committed:
Fixed all the lint errors

jatinkksharma committed:
Merge pull request #9 from brightscout-alpha/MI-1137

MI-1137 - Misc components

jatinkksharma committed:
update custom emoji component

jatinkksharma committed:
Merge branch 'MI-1137' of github.com:brightscout-alpha/mattermost-webapp into MI-1136

jatinkksharma committed:
fix emoji overlay position

jatinkksharma committed:
add custom status in user post

jatinkksharma committed:
update logic for custom emoji component

jatinkksharma committed:
update css for custom status post header

jatinkksharma committed:
add setting to enable disble custom status

jatinkksharma committed:
fix ui for user post component

jatinkksharma committed:
misc review fixes

jatinkksharma committed:
add custom status enabled check in post header

chetanyakan committed:
Use i18n texts

chetanyakan committed:
Merge pull request #7 from brightscout-alpha/MI-1136

Added custom status in user profile popover

chetanyakan committed:
MI-1140 - Added custom status tooltip (#10)

* Added conditional tooltip everywhere where the custom status text is overflowing
* Changed set status in profile popover from anchor to button

manojmalik20 committed:
Implementing opening status dropdown from post header
Made redux action, reducer and selector for opening status dropdown
Used the redux store variable to open menu
Added conditions for the Update status and pulsating dot rendering
Added condition for pulsating dot
Added new redux logic for opening status dropdown by clicking Update status button
Added new logic for showing Update status button on the last post by the current user
Optimized the GetCurrentUserLastGroupedPostId util
Changed the name of the few of the functions and variables

chetanyakan committed:
MI-1141: Add pulsating dot component (#11)

chetanyakan committed:
MI-1141: Add pulsating dot component (#11)

manojmalik20 committed:
Fixed the bug in CurrentUserLastPost util and made utils for Postheader and CustomStatusPulsating dot

chetanyakan committed:
Update CSS, Fix lint

manojmalik20 committed:
Conditional rendering of the pulsating dot
Modified userProps to store object instead of string in the initialProps
Modified the action to use new initialProps type
Fixed lint errors

chetanyakan committed:
code refactoring

manojmalik20 committed:
Merge branch 'MI-1141_3' of github.com:brightscout-alpha/mattermost-webapp into MI-1141_3

manojmalik20 committed:
Merge branch 'MI-1141_3' of github.com:brightscout-alpha/mattermost-webapp into MI-1141_3

chetanyakan committed:
Merge branch 'custom-status' of github.com:brightscout-alpha/mattermost-webapp into MI-1141_3

* 'custom-status' of github.com:brightscout-alpha/mattermost-webapp:
  MI-1141: Add pulsating dot component (#11)

chetanyakan committed:
Merge branch 'MI-1141_3' of github.com:brightscout-alpha/mattermost-webapp into MI-1141_3

* 'MI-1141_3' of github.com:brightscout-alpha/mattermost-webapp:
  Conditional rendering of the pulsating dot Modified userProps to store object instead of string in the initialProps Modified the action to use new initialProps type Fixed lint errors

chetanyakan committed:
fix css

manojmalik20 committed:
Fixed the bug in pulsating dot and changed the naming of some variables

manojmalik20 committed:
Changed the name of several variables

manojmalik20 committed:
Added snapshot testing of the custom status components

manojmalik20 committed:
WIP: Post header CSS formatting

manojmalik20 committed:
Modified the conditions for the pulsating dot and the Post header
Fixed the lint error in the MenuWrapper

manojmalik20 committed:
Fixed the Pulsating dot and Post header condition util
Fixed the DM channel old bug
Changed the name of the util
Fixed the menu wrapper getDerivedStateFromProps

manojmalik20 committed:
Fixed the condition in getDerivedStateFromProps in MenuWrapper component

manojmalik20 committed:
Added some more tests for custom status components
Added tests for status dropdown and custom status selectors

chetanyakan committed:
Refactor CSS

manojmalik20 committed:
WIP : Test cases for the status dropdown component
Completed test cases for the selectors

manojmalik20 committed:
Fixed the coloring for the Post header Custom status link

manojmalik20 committed:
Merge branch 'MI-1141_4' of github.com:brightscout-alpha/mattermost-webapp into MI-1142

manojmalik20 committed:
Complete test cases of status dropdown
Complete test cases of profile popover
Complete test cases of menu wrapper
WIP test cases of post related components

manojmalik20 committed:
Replaced custom status actions with mattermost redux actions

manojmalik20 committed:
Added post header and pulsating dot showing logic in the profile popover set status

manojmalik20 committed:
Merge branch 'MI-1141_5' of github.com:brightscout-alpha/mattermost-webapp into MI-1142

manojmalik20 committed:
Modified custom status modal to handle initialization state

manojmalik20 committed:
Merge branch 'MI-1141_5' of github.com:brightscout-alpha/mattermost-webapp into MI-1142

manojmalik20 committed:
Added test cases for several components
Added test cases for channel header
Added test cases for sidebar channel button or link
Added test cases for post header
Fixes in status dropdown and profile popover tests

manojmalik20 committed:
Added tests for status dropdown action

manojmalik20 committed:
Fixed the enforceFocus problem in generic modal and custom status modal

manojmalik20 committed:
Post header and pulsating dot conditions(#15)

* WIP: Post header CSS formatting

* Modified the conditions for the pulsating dot and the Post header
Fixed the lint error in the MenuWrapper

* Fixed the Pulsating dot and Post header condition util
Fixed the DM channel old bug
Changed the name of the util
Fixed the menu wrapper getDerivedStateFromProps

* Fixed the condition in getDerivedStateFromProps in MenuWrapper component

* Refactor CSS

* Fixed the coloring for the Post header Custom status link

* Replaced custom status actions with mattermost redux actions

* Added post header and pulsating dot showing logic in the profile popover set status

* Modified custom status modal to handle initialization state

* Modified the getCurrentUserLastPostGroupFirstPostId util to make it simple

* Changed names of some variables and fixed some issues
Fixed RenderEmoji issue
Fixed Post util issue

Co-authored-by: Chetanya Kandhari <chetanya.kandhari@brightscout.com>

manojmalik20 committed:
Test Cases (#14)

* MI-1141: Add pulsating dot component (#11)

* Implementing opening status dropdown from post header
Made redux action, reducer and selector for opening status dropdown
Used the redux store variable to open menu
Added conditions for the Update status and pulsating dot rendering
Added condition for pulsating dot
Added new redux logic for opening status dropdown by clicking Update status button
Added new logic for showing Update status button on the last post by the current user
Optimized the GetCurrentUserLastGroupedPostId util
Changed the name of the few of the functions and variables

* Fixed the bug in CurrentUserLastPost util and made utils for Postheader and CustomStatusPulsating dot

* Update CSS, Fix lint

* Conditional rendering of the pulsating dot
Modified userProps to store object instead of string in the initialProps
Modified the action to use new initialProps type
Fixed lint errors

* code refactoring

* fix css

* Fixed the bug in pulsating dot and changed the naming of some variables

* Changed the name of several variables

* Added snapshot testing of the custom status components

* WIP: Post header CSS formatting

* Modified the conditions for the pulsating dot and the Post header
Fixed the lint error in the MenuWrapper

* Fixed the Pulsating dot and Post header condition util
Fixed the DM channel old bug
Changed the name of the util
Fixed the menu wrapper getDerivedStateFromProps

* Fixed the condition in getDerivedStateFromProps in MenuWrapper component

* Added some more tests for custom status components
Added tests for status dropdown and custom status selectors

* Refactor CSS

* WIP : Test cases for the status dropdown component
Completed test cases for the selectors

* Fixed the coloring for the Post header Custom status link

* Complete test cases of status dropdown
Complete test cases of profile popover
Complete test cases of menu wrapper
WIP test cases of post related components

* Replaced custom status actions with mattermost redux actions

* Added post header and pulsating dot showing logic in the profile popover set status

* Modified custom status modal to handle initialization state

* Added test cases for several components
Added test cases for channel header
Added test cases for sidebar channel button or link
Added test cases for post header
Fixes in status dropdown and profile popover tests

* Added tests for status dropdown action

* Fixed the enforceFocus problem in generic modal and custom status modal

Co-authored-by: Chetanya Kandhari <chetanya.kandhari@brightscout.com>

manojmalik20 committed:
Merge branch 'custom-status' of github.com:brightscout-alpha/mattermost-webapp into custom-status

manojmalik20 committed:
Merge branch 'MI-1141_3' of github.com:brightscout-alpha/mattermost-webapp into MI-1141_3

chetanyakan committed:
Merge pull request #12 from brightscout-alpha/MI-1141_3

Pulsating dot and post header components

manojmalik20 committed:
Fixed the bug in Post header component

manojmalik20 committed:
Added Markdown component for the custom status text

harshilsharma63 commented on actions/views/custom_status.ts:
```diff
@@ -0,0 +1,30 @@
+export function clearCustomStatusInitialisationState() {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const user = {...getCurrentUser(getState())};
+        const userProps = {...user.props};
```
Any reason we can do `user.props.customStatusInitialisationState = ''`? Is it to forcefully create a new object?

harshilsharma63 commented on components/custom_status/custom_status.scss:
```diff
@@ -0,0 +1,181 @@
+.StatusModal {
+    width: 600px !important;
```
Any way to avoid `!important` here?

manojmalik20 replied:
This is to override the styles of another class. Added this comment in the code to explain.

harshilsharma63 commented on components/status_dropdown/status_dropdown.jsx:
```diff
@@ -116,10 +171,101 @@ export default class StatusDropdown extends React.PureComponent {
 
+        let customStatusComponent;
+        if (this.props.isCustomStatusEnabled) {
+            let customStatusText = localizeMessage('status_dropdown.set_custom', 'Set a Custom Status');
```
We can move this code block in a separate function to make if block cleaner.

harshilsharma63 commented on utils/post_utils.jsx:
```diff
@@ -486,3 +486,56 @@ export function makeGetReplyCount() {
+        !isSystemMessage(post) && !isSystemMessage(previousPost); // And neither is a system message
+}
+
+export function getPostFromId(state, id) {
```
This function name sounds like a generic function for fetching post by ID, but it does more than that. Can this be renamed?

manojmalik20 committed:
Added emoji handling in custom status text everywhere and review fixes

chetanyakan committed:
Refactor custom status emoji

manojmalik20 committed:
Merge pull request #16 from brightscout-alpha/MI-1045_1

Refactor custom status profile popover component

manojmalik20 committed:
Merge branch 'custom-status' of github.com:brightscout-alpha/mattermost-webapp into MI-1148

manojmalik20 committed:
Added Markdown component in channel header and profile popover
Added custom status text limit

manojmalik20 committed:
Added tootip and fixed the width of the DM header custom status component
Changed the custom status text character limit
Changed the names of several variables

chetanyakan committed:
Merge pull request #17 from brightscout-alpha/MI-1148

chetanyakan committed:
Merge pull request #18 from brightscout-alpha/MI-1148

Added tootip and fixed the width of the DM header custom status compo…

chetanyakan committed:
Custom status base (#19)

* add new menu group and menu item to set custom status

* show modal on click of set custom status menu item

* Add custom status input modal to show on menu item click

* add consyand id for custom status modal

* Updating UI for custom status

* call redux action to set custom status

* show status in profile popover

* fix dictionary key

* show custom status only if present

* Store emoji and expiration date in state

* handle props changed web socket events

* use patch me method to update props as it already broadcasts props

* show custom status emoji besides username in header

* move status up in profile popover

* show custom status message in dropdown menu

* Revert "handle props changed web socket events"

This reverts commit dab1c9927cfe1e25c80c707f7d73ab6b30a14729.

* maintain recent 5 statuses in a list in props

* show status emoji besides name

TODO: currently it renders name of emoji instead of emoji itself so fix it

* fix typo

* append ... only if status is longer than 24 chars in header dropdown

Co-authored-by: Umang Parmar <umangjparmar@gmail.com>
Co-authored-by: Asaad Mahmood <asaadmahmood@users.noreply.github.com>

chetanyakan committed:
Revert "Custom status base (#19)"

This reverts commit 401ae1e7bfcbb8f310223f1d66143fb4bd193c5e.

chetanyakan committed:
Merge branch 'custom-status-1' into custom-status

* custom-status-1: (153 commits)
  Revert "Custom status base (#19)"
  Custom status base (#19)
  fix/update failed E2E tests (#7430)
  [MM-31791] Support Packet Generation FRONTEND (#7314)
  [MM-32269] System Console: TeamIcon edge cases (#7395)
  update steps for google openid (#7327)
  git checkout -b fix-slash-commands-tm4j (#7397)
  MM-31646: Update Office instructions (#7326)
  Cypress/E2E: Fix tests for archived channels (#7424)
  Cypress/E2E: Fix tests for bots in list and in sidebar (#7421)
  [MM-29108] Simplifies emoji reaction animation (#7386)
  Cypress/E2E: Overwrite cy.visit and cy.reload to have default 3s wait (#7410)
  [MM-22369] Filter out ldap group sync teams when adding a team to a user (#7324)
  MM 31192  - license renewal telemetry (#7413)
  Migrate message_attachments folder to Typescript (#7339)
  Cypress/E2E: Fix and reorganize tests for custom emojis (#7415)
  promote tests with 23 tm4j test cases (#7402)
  fix failed tests due to recent change in add users modal (#7398)
  fix for more channels and auth specs (#7416)
  fix e2e for sidebar channels (#7412)
  ...

chetanyakan committed:
Merge branch 'custom-status' of github.com:brightscout-alpha/mattermost-webapp into custom-status

* 'custom-status' of github.com:brightscout-alpha/mattermost-webapp:
  Added tootip and fixed the width of the DM header custom status component Changed the custom status text character limit Changed the names of several variables

manojmalik20 committed:
Added Custom status emoji in rhs posts (#21)

* Added Custom status emoji in rhs posts
Modified the getCurrentUserLastPostGroupFirstPostId util to handle comments and replies

* Review fixes
Moved Enable custom status config setting to last of the page
Modified custom status modal to show only those suggestions that are not in the recent list
Fixed CSS of the custom status modal

* Review and post rebase fixes
Optimized the post list not to find currentUserLastPost if Update status post header is not visible
Fixed the styling of the channel header custom status
Fixed the coloring of the Set status menu option in status dropdown
Fixed the styling of the custom status modal
Fixed the styling of the custom status tooltip everywhere
Fixed the styling of the Update your status post header
Added custom status emoji in the sidebar channel where new component is used after rebase
Fixed the styling of the new emoji added in the sidebar channel list

* Typecheck fixes and updated snapshots

* Modified the getLastCurrentUserPostWithHeaderId util for better readability

manojmalik20 committed:
Fix lint

manojmalik20 committed:
CSS Fixes (#22)

* Added Custom status emoji in rhs posts
Modified the getCurrentUserLastPostGroupFirstPostId util to handle comments and replies

* Review fixes
Moved Enable custom status config setting to last of the page
Modified custom status modal to show only those suggestions that are not in the recent list
Fixed CSS of the custom status modal

* Review and post rebase fixes
Optimized the post list not to find currentUserLastPost if Update status post header is not visible
Fixed the styling of the channel header custom status
Fixed the coloring of the Set status menu option in status dropdown
Fixed the styling of the custom status modal
Fixed the styling of the custom status tooltip everywhere
Fixed the styling of the Update your status post header
Added custom status emoji in the sidebar channel where new component is used after rebase
Fixed the styling of the new emoji added in the sidebar channel list

* Typecheck fixes and updated snapshots

* Modified the getLastCurrentUserPostWithHeaderId util for better readability

chetanyakan committed:
Update package-lock.json

chetanyakan committed:
Merge branch 'master' of github.com:mattermost/mattermost-webapp into custom-status

* 'master' of github.com:mattermost/mattermost-webapp:
  [MM-32120] - Restrict calls to `/cloud/subscription` endpoint to only admins (#7437)
  Cypress/E2E: Fix channel switch spec (#7472)
  Cypress/E2E: Split file preview e2e tests (#7454)
  Cypress/E2E: Added linux platform (#7466)
  Cypress/E2E: Move subpath login under subpath folder (#7465)
  Cypress/E2E: Potential fix for user management (#7464)
  Cypress/E2E: Fix dm indicator in title spec (#7463)
  Add beta to bg (#7455)
  [MM-32121] Fix neverending load on one-paged search results (#7449)
  Cypress/E2E: Fix misc failing tests (#7452)
  Rm nm from cache (#7453)
  MM- 18893: Change text User to Member (#7323)
  [MM-24635] adds an Input component with max length validation and visual feedback (#7281)
  fix console warning (#7450)
  Cypress/E2E: Fix MM-T1752 enable account creation (#7429)
  Translations update from Weblate (#7445)

manojmalik20 committed:
Feedback suggestions (#23)

* Fixed the review issues by Mattermost
Fixed the SCSS structure of several components - Generic Modal, custom status, post header, status dropdown
Refactoring of code in several components
Added Menu.ItemToggleModalRedux in the set custom status option in the Status dropdown

* Fixed button descendant of button error and some review fixes
Refactored the showTooltip code in all files
Changed the customStatusTextLimit
Fixed passing empty object in the emojiStyle prop in CustomStatusEmoji

* Made a separate component for custom status text (#24)

Modified channel header to use customStatusText component
Modified status dropdown to use customStatusText component
Modified profile popover to use customStatusText component
Modified customStatusSuggestion to use customStatusText component

chetanyakan committed:
Merge branch 'master' of github.com:mattermost/mattermost-webapp into custom-status

* 'master' of github.com:mattermost/mattermost-webapp:
  MM-31888: adjusted padding on menu items in sidebar (#7462)
  MM-32226: adjustted "more unreads"-indicator positioning (#7458)
  Translations update from Weblate (#7498)
  use auto height (#7425)
  [MM-18952/GH-16740] - Migrate 'components/authorize' module and associated tests to TypeScript (#7427)
  MM-31916: channel mention for specific keywords breaks application (#7493)
  Cypress/E2E: Fix more unreads position with scroll spec (#7485)
  MM-32131: Fix header alignment when channel has guests (#7381)
  [MM-32581][MM-32582] - Fix popover regressions (#7480)
  Cypress/E2E: Fix team permissions spec (#7486)
  Cypress/E2E: Fix search box focus related failures (#7477)
  MM-20403 Migrate mfa/setup to typescript (#7353)
  [MM-32401][MM-32450] Some style fixes for the sidebar (#7451)
  added `channel.name` to the concatenated searchString (#7476)
  MM-32558: changed SidebarHeader class to span full-width (#7459)
  Cypress/E2E: Fix channel settings spec (#7471)
  Cypress/E2E: Separated bot accounts in lists spec (#7470)
  Removing the preview limit for PDF files (#7345)

chetanyakan committed:
Fix lint

manojmalik20 committed:
Modified the custom status initialization state logic to preferences … (#25)

* Modified the custom status initialization state logic to preferences instead of user props

* Fixed the button and li issue reported in the review (#26)

* Fixed the button and li issue reported in the review
Moved the setCustomStatusInitializationState action to the redux repo
Updated some snapshots and deleted some tests

* Moved recentCustomStatuses logic to preferences
Modified getRecentCustomStatuses selector to use preferences
Modified CustomStatusModal to use getRecentCustomStatuses selector
Added tests for CustomStatusText component

* Updated the tests for Custom Status Modal

* Moved the showPulsatingDotAndPostHeader logic in the selectors and wrote tests for them

chetanyakan committed:
Merge branch 'master' of github.com:mattermost/mattermost-webapp into custom-status

* 'master' of github.com:mattermost/mattermost-webapp:
  Cypress/E2E: Fix slash commands spec and some promotions (#7522)
  E2 e prod tests fixes (#7509)
  MM-31503 - screen to restrict access because workspace capacity" (#7468)
  [MM-20378] Convert file_attachment_list and tests to TypeScript (#7488)
  [MM-20377] Convert persist_gate to TypeScript  (#7434)
  [MM-32301] Post Aria Label (#7457)
  [MM-30042]: CWS - Step 3 of Getting Started shows error when trying to send 10 email addresses (#7439)
  fix typo (#7505)
  Add images for email notification redesign (#7417)
  Cypress/E2E: Distribute tests evenly when testing in parallel (#7508)
  Cypress/E2E: Use MinIO for S3 storage testing (#7506)
  MM-32354 - do not show renew now text in announcement bar tooltip (#7491)
  fix for test that requires untrusted connection (#7507)
  Revert "Allow uploading 10 files per post (#7160)" (#7510)
  MM-26497: markdown table word breaking (#7500)
  Cypress/E2E: Promote/Demote e2e tests (#7497)
  [MM-31497] - Prevent end users from inviting people and exceeding the free tier limits (#7408)
  MM-31961: fixed color in search field for dark-theme (#7490)

manojmalik20 committed:
Fixed the review bugs and emoji overlay placement bug (#29)

* Fixed the review bugs and emoji overlay placement bug
Added the functionality to open custom status modal by clicking the custom status emoji in the sidebar header
Fixed the positioning of the custom status in the status dropdown
Fixed the emoji overlay positioning bug in custom status modal when zooming in more than 110% or in small resolutions

* Fixed the type of the defaultHorizontalPosition in the popOverOverlay util

* Fixed review comments
Changed the openModalOnClick functionality to take generic onClick as prop in the CustomStatusEmoji
Modified the emoji picker overlay to take props to handle right position of the overlay

* Modified the function for custom status emoji click in sidebar header

manojmalik20 committed:
Apply suggestions from code review

Co-authored-by: Harrison Healey <harrisonmhealey@gmail.com>

manojmalik20 committed:
Review fixes (#30)

Modified and added QuickInput component in Custom Status Modal
Modified the CSS for the custom status input and Status Dropdown clear button
Fix formatted message for i18n
Made separate component for PostHeaderCustomStatus
Changed prop in RenderEmoji from emoji to emojiName
Changed prop in emojiPicker from autoComplete to autocomplete
Removed the currentUserLastPostGroupFirstPost logic from all components
Modified the setStatusDropdown action not to use redux-thunk
Updated custom status components test cases to not include the Provider in the snapshots
CSS fixes
Added localization in the custom status default suggestions
Added localization ids in the i18n/en.json
Modified and added QuickInput component in Custom Status Modal
Modified the CSS for the custom status input and Status Dropdown clear button

Co-authored-by: Chetanya Kandhari <chetanya.kandhari@brightscout.com>

chetanyakan committed:
Merge branch 'master' of github.com:mattermost/mattermost-webapp into custom-status

* 'master' of github.com:mattermost/mattermost-webapp:
  Translations update from Weblate (#7528)
  [MM-20396] Convert copy_url_context_menu to TypeScript (#7487)
  MM-22267: line-breaks in inline images are now correctly rendered (#7502)
  Cypress/E2E: Fix webhook spec zephyr keys (#7524)
  [MM-20420]Migrate 'components/add_groups_to_channel_modal' module and associated tests to TypeScript (#7176)

manojmalik20 committed:
Review fixes by MM (#31)

Memoized the getCustomStatus and getRecentCustomStatuses selector
Changed the name of parameters in position utils and modified handling topOffset in emoji picker tabs
Removed the dialogProps from the custom status modal opening function everywhere
Removed the use of strong tags and used span with CSS
Added localization in the custom status modal and respective ids in i18n/en.json
Added placeholder in the quick input component
Updated snapshot of Profile popover and tests of custom status selectors
Fixed type-check issues and the width issue in the LHS channel name
Added remaining localization ids in en.json
Modified showing update status button logic only via CSS and fixed the bug of the header not showing on comments
Changed names of some classes like bold
Changed names of some variables

manojmalik20 committed:
Modified the custom status initialization state selectors and the act… (#32)

* Modified the custom status initialization state selectors and the action call

* Update selectors/views/custom_status.ts

Co-authored-by: Chetanya Kandhari <chetanya.kandhari@brightscout.com>

Co-authored-by: Chetanya Kandhari <chetanya.kandhari@brightscout.com>

chetanyakan committed:
Changes from review, fix types (#33)

* Changes from review, fix types

* Fix scss

* Fix type

chetanyakan committed:
Feedback changes and CSS tweaks

* Feedback changes and CSS tweaks

* Update snapshots

chetanyakan committed:
Fix lint

chetanyakan committed:
Fix RHS CSS

chetanyakan committed:
Update selectors/views/custom_status.ts

Co-authored-by: Harrison Healey <harrisonmhealey@gmail.com>

chetanyakan committed:
Fix lint

chetanyakan committed:
Fix RHS post CSS

chetanyakan committed:
Fix i18n

chetanyakan committed:
Merge branch 'master' of github.com:mattermost/mattermost-webapp into custom-status

* 'master' of github.com:mattermost/mattermost-webapp:
  MM-27400: tooltip for long channel names (#7527)
  Remove output from pkill (#7536)
  MM-27241 Private channel prompt for sysadmin (#7489)
  [MM-32087] - Add theme variable for sidebar team bar (#7389)
  MM-32825 - suggestion box hidden by rhs (#7523)
  MM-32928 Remove custom-protocol-detection (#7532)
  [MM-32684] Fix RHS Autocomplete options cover the input box for some reply threads (#7504)
  MM-3190 show user limit modal when cloud limit exceed (#7496)
  MM-28736 Update dependencies (#7512)
  MM-31501 - Add upgrade-now link for admins in invite users (#7446)

chetanyakan committed:
Update package.json

chetanyakan committed:
Fix types

chetanyakan committed:
Update snapshots

manojmalik20 committed:
Merge branch 'master' of github.com:brightscout-alpha/mattermost-webapp into custom-status

manojmalik20 committed:
Merge branch 'master' of github.com:brightscout-alpha/mattermost-webapp into custom-status

manojmalik20 committed:
Fixed tests (#36)

Fixed the tests for the custom status selectors
Fixed the tests for the searchable channel list

manojmalik20 committed:
Removed unused utils and fix the remaining tests

chetanyakan committed:
Update snapshots

chetanyakan commented:
Closing as merged to MM via another PR.
